### ISSUE TITLE
AdaptiveAvgPool2d

### ISSUE THREAD

**Issue by shrutishrestha** [VainF/DeepLabV3Plus-Pytorch#27]
Why is this nn.AdaptiveAvgPool2d(1) done here?

class ASPPPooling(nn.Sequential):
   
    def __init__(self, in_channels, out_channels):
        super(ASPPPooling, self).__init__(
            nn.AdaptiveAvgPool2d(1),
            nn.Conv2d(in_channels, out_channels, 1, bias=False),
            nn.BatchNorm2d(out_channels),
            nn.ReLU(inplace=True))

    def forward(self, x):
        size = x.shape[-2:]
        x = super(ASPPPooling, self).forward(x)
        return F.interpolate(x, size=size, mode='bilinear', align_corners=False)

I am doing segmentation task and this abive pooling changes my output from torch.Size([1, 256, 16, 16]) to torch.Size([1, 256, 1, 1])
giving the error,
"Expected more than 1 value per channel when training, got input size torch.Size([1, 256, 1, 1])"

What could have gone wrong?

**Comment by Nuller-CV** [comment]
I have the same problem. How did you solve it？

**Comment by shrutishrestha** [comment]
> I have the same problem. How did you solve it？

Hi, how do you want your output? The line "nn.AdaptiveAvgPool2d(1)" means 1 is the output size and your output will be of height 1 and width 1.

**Comment by VainF** [comment]
BatchNorm requires at least 2 values per channel for mean and variance estimation. So,  in training mode, please make sure that your batch_size is larger than 1.

**Comment by DISAPPEARED13** [comment]
what if batch_size is 1, can I just replace `nn.BatchNorm` with `nn.InstanceNorm`?
